In [115]:
import numpy as np
from scipy import constants 
import math

In [116]:
c=constants.c
e=constants.e

In [156]:
length=6000 #meters
dipole_length=14 #meters
fill_factor=0.67
number_dipols=length*0.67/dipole_length
number_dipols=math.ceil(number_dipols)
print('Total number of dipols:',number_dipols)

Total number of dipols: 288


<img src="./figures/LHC_cell.jpeg" width="600">

In [158]:
fodo_cell_length=100 #meter
number_fodo_cells=length/fodo_cell_length
quad_driftspace=fodo_cell_length/2.
print('Distance between two quadrupols:',quad_driftspace,'meter')

Distance between two quadrupols: 50.0 meter


In [120]:
dipols_per_driftspace=3
dipols_per_cell=2*dipols_per_driftspace # -> 6 dipols per cell

In [160]:
cells_with_dipols=number_dipols/dipols_per_cell
print('Number of cells containing dipols:',cells_with_dipols)

Number of cells containing dipols: 48.0


In [161]:
cells_straight_sections=number_fodo_cells-cells_with_dipols
print('Number of cells in straight sections:',cells_straight_sections)

Number of cells in straight sections: 12.0


In [162]:
fill_factor_in_fodo=6*dipole_length/(2*quad_driftspace)
print('Dipole fill factor in dipole FODO cells:',fill_factor_in_fodo*100,'%')

Dipole fill factor in dipole FODO cells: 84.0 %


<img src="./figures/fodo_wiedemann.jpeg" width="600">

- Stability: $|f|> \frac{L}{2}$ with $L$ the drift space between two quads 

- Optimal phase advance for round beam (e.g. protons):
$\Psi\overset{!}{=}90^\circ$
$\rightarrow f\overset{!}{=}\sqrt{2}L$

In [159]:
focal_length=quad_driftspace*np.sqrt(2)
print('Focal distance:',focal_length,'meter')

Focal distance: 70.71067811865476 meter


In [164]:
phase_advance=np.arccos(1-2*quad_driftspace**2/focal_length**2)
print('Phase advance:',phase_advance/(2*np.pi)*360,'degree')

Phase advance: 90.0 degree


### Quadrupole
- Gradient: $g=\frac{2\mu_0 n I}{R^2}$
- Strength: $k=\frac{e}{p}g=0.2998\frac{g[\mathrm{T/m}]}{p[\mathrm{Gev/c}]}$
- Focal length $f=\frac{1}{k\cdot l}$ with $l$ length of the quad

In [165]:
energy=3836/2. #GeV

In [166]:
quad_length= 1 # meter (LHC: 3.1 meter magnetic length)
quad_strength=1./(focal_length*quad_length)
print('Quadrupole strength k:',quad_strength,'m^-2')

Quadrupole strength k: 0.01414213562373095 m^-2


In [167]:
quad_gradient=quad_strength*energy/0.2998 
print('Quadrupole gradient g:',quad_gradient,'T/m (LHC: 220 T/m)')

Quadrupole gradient g: 90.47570422386912 T/m (LHC: 220 T/m)


In [168]:
quad_coil_current = 10000 # Ampere (LHC: 11470A)
quad_n_turns = 20

In [170]:
quad_aperture=np.sqrt(2*constants.mu_0*quad_n_turns*quad_coil_current/quad_gradient)
print('Quadrupole aperture R:',quad_aperture*1000,'mm (LHC: 56mm)')

Quadrupole aperture R: 74.536490719232 mm (LHC: 56mm)


In [171]:
mat_fodo=np.matrix([[0,2*quad_driftspace*(1+quad_driftspace/focal_length)],[-2*(1-quad_driftspace/focal_length)*quad_driftspace/(focal_length**2),0]])
mat_fodo

matrix([[ 0.00000000e+00,  1.70710678e+02],
        [-5.85786438e-03,  0.00000000e+00]])

$\beta=\beta^+$ in centre of the (horizontally) focusing quadrupole

$\beta=\beta^-$ in centre of the (horizontally) defocusing quadrupole


In [147]:
fodo_parameter=focal_length/quad_driftspace

In [174]:
beta_plus=quad_driftspace*fodo_parameter*(fodo_parameter+1)/np.sqrt(fodo_parameter**2-1)
beta_plus_opt=quad_driftspace*(2+np.sqrt(2))
print('beta plus (=optimal/minimal beta plus): ',beta_plus,'meter')

beta plus (=optimal/minimal beta plus):  170.71067811865473 meter


In [175]:
beta_minus=quad_driftspace*fodo_parameter*(fodo_parameter-1)/np.sqrt(fodo_parameter**2-1)
beta_minus_opt=quad_driftspace*(2-np.sqrt(2))
print('beta minus: ',beta_minus,'meter')

beta minus:  29.28932188134525 meter


Comparison with LHC:
- Phase advance $90^°$
- $\beta^+\approx 180\,$m
- $\beta^-\approx 30\,$m

In [176]:
emittance=2.5e-9 #meter rad
beam_envelope_max=np.sqrt(emittance*beta_plus_opt)
print('Maximum beam envelope with emittance of 2.5 nm rad:',beam_envelope_max*1000,'mm')

Maximum beam envelope with emittance of 2.5 nm rad: 0.6532814824381883 mm


$\rightarrow$ more than enough space in quadrupole aperture ($74.5$ mm) to fit a beam screen of the needed 10 sigma of the beam size: $(10\cdot 0.65)$ mm